SGAPI/ItemSets/SGItemSet.m

```diff
@@ -89,7 +89,16 @@ - (void)cancelFetch {
 - (void)processResults:(NSData *)data {
     __weakSelf me = self;
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+        NSError *error = nil;
+        NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+        if (error) {
+            if (me.onPageLoadFailed) {
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    me.onPageLoadFailed(error);
+                });
+            }
+            return;
```

Comment:
hmm, well not necessarily a bug, since `onPageLoaded` also seems like the wrong thing to call in this case. I guess if a user of the library hasn't defined both of these handler properties, then they also probably don't expect one to always get called, so it's fine as-is.